src/NServiceBus.Callbacks/ExtendableOptionsExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace NServiceBus
+{
+    using System.Threading;
+    using NServiceBus.Extensibility;
+
+    static class ExtendableOptionsExtensions
+    {
+        public static void RegisterTokenInternal(this ExtendableOptions options, CancellationToken cancellationToken)
+        {
+            var extensions = options.GetExtensions();
+            UpdateRequestResponseCorrelationTableBehavior.RequestResponseParameters data;
```

Comment:
Not regarding this PR, but in general I think we need to provide `GetOrCreate` convenience method on the extensions object because we will have this kind of code everywhere.